src/view/SignupView.java

```diff
@@ -0,0 +1,141 @@
+package view;
+
+import interface_adapter.signup.SignupController;
+import interface_adapter.signup.SignupState;
+import interface_adapter.signup.SignupViewModel;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+
+public class SignupView extends JPanel implements ActionListener, PropertyChangeListener{
+
+    public static String viewName = "Signup Page";
+
+    private final SignupViewModel signupViewModel;
+    private final JTextField usernameInputField = new JTextField(15);
+    private final JPasswordField passwordInputField = new JPasswordField(15);
+    private final JPasswordField repeatPasswordInputField = new JPasswordField(15);
+    private final SignupController signupController;
+    private final JButton signUp;
+
+    public SignupView(SignupController controller, SignupViewModel viewModel) {
+
+        this.signupController = controller;
+        this.signupViewModel = viewModel;
+        viewModel.addPropertyChangeListener(this);
+
+        JLabel title = new JLabel(SignupViewModel.TITLE_LABEL);
+        JPanel buttons = new JPanel();
+        signUp = new JButton(SignupViewModel.SIGNUP_BUTTON_LABEL);
+        buttons.add(signUp);
+
+        LabelTextPanel usernameInfo = new LabelTextPanel(
+                new JLabel(SignupViewModel.USERNAME_LABEL), usernameInputField);
+        LabelTextPanel passwordInfo = new LabelTextPanel(
+                new JLabel(SignupViewModel.PASSWORD_LABEL), passwordInputField);
+        LabelTextPanel repeatPasswordInfo = new LabelTextPanel(
+                new JLabel(SignupViewModel.REPEAT_PASSWORD_LABEL), repeatPasswordInputField);
+
+
+        signUp.addActionListener(
+            this
+        );
+
+        usernameInputField.addKeyListener(
+                new KeyListener() {
+                    @Override
+                    public void keyTyped(KeyEvent e) {
+                        SignupState currentState = signupViewModel.getState();
+                        String text = usernameInputField.getText() + e.getKeyChar();
+                        currentState.setUsername(text);
+                        signupViewModel.setState(currentState);
+                    }
```

Comment:
below here and in the passwordInputField as well as repeatPasswordInputField, there are these keyPressed methods that don't seem to do anything?